prov/util/src/util_attr.c

```diff
@@ -1002,6 +1002,92 @@ int ofi_check_tx_attr(const struct fi_provider *prov,
 	return 0;
 }
 
+/*
+ * Check user do not request attr more restrictive than what prov can offer
+ */
+static bool ofi_compare_hmem_attr_opt(enum fi_hmem_attr_opt prov_opt,
+				  enum fi_hmem_attr_opt user_opt)
+{
+	switch (user_opt) {
+	case FI_HMEM_ATTR_ENABLED:
+		return (prov_opt == FI_HMEM_ATTR_ENABLED ||
+			prov_opt == FI_HMEM_ATTR_PREFERRED ||
+			prov_opt == FI_HMEM_ATTR_REQUIRED ||
+			prov_opt == FI_HMEM_ATTR_DISABLED);
+	case FI_HMEM_ATTR_REQUIRED:
+		return (prov_opt == FI_HMEM_ATTR_REQUIRED);
```

Comment:
When user request a required and preferred, we should fail only when prov_opt is FI_HMEM_ATTR_DISABLED, right?